platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -210,60 +210,52 @@ Startup Tasks:
 
 Cleanup Tasks:
 
-1. Once the first saved state is created, All `config.txt` related files are moved to subdirectory `.
-   archive/yyyy-MM-dd_HH-mm-ss/`.
+1. Once the first saved state is created, All `config.txt` related files are moved to subdirectory
+   `.archive/yyyy-MM-dd_HH-mm-ss/`.
 
-##### MODE: Software Upgrade
+##### MODE: Migration Software Upgrade
 
 Conditions:
 
 1. A state is loaded from disk
 2. A software upgrade is happening
-3. No `override-config.txt` file is present on disk
+3. A `config.txt` file is present on disk
+4. The roster state is empty (no candidate roster and no active rosters)
+5. No `override-config.txt` file is present on disk
 
 Ignored Files: (Log warning if present)
 
 1. `genesis-config.txt`
-2. `config.txt` - if a candidate roster is present in the state
 
 Startup Tasks:
 
-1. If there is a candidate roster in the state (Non-Migration Software Upgrade)
-   1. candidateRoster := read the candidate roster from the roster state.
-   2. currentRound := state round +1
-   3. (previousRoster, previousRound) := read the latest (current) active roster and round from the roster state.
-   4. new rosterHistory := `[(candidateRoster, currentRound), (previousRoster, previousRound)]`
-   5. clear the candidate roster from the roster state.
-   6. set (candidateRoster, currentRound) as the new active roster in the roster state.
-   7. Start the platform with the new rosterHistory.
-2. If the roster state is empty: no candidate roster and no active rosters. (Migration Software Upgrade)
-   1. Read the current AddressBooks from the platform state.
-   2. configAddressBook := Read the address book in config.txt
-   3. previousRoster := translateToRoster(currentAddressBook)
-   4. currentRoster := translateToRoster(configAddressBook)
-   5. currentRound := state round +1
-   6. rosterHistory := `[(currentRoster, currentRound), (previousRoster, 0)]`
-   7. set (previousRoster, 0) as the active roster in the roster state.
-   8. set (currentRoster, currentRound) as the active roster in the roster state.
-   9. Start the platform with the new rosterHistory.
+1. Read the current AddressBooks from the platform state.
+2. configAddressBook := Read the address book in config.txt
+3. previousRoster := translateToRoster(currentAddressBook)
+4. currentRoster := translateToRoster(configAddressBook)
+5. currentRound := state round +1
+6. rosterHistory := `[(currentRoster, currentRound), (previousRoster, 0)]`
+7. set (previousRoster, 0) as the active roster in the roster state.
+8. set (currentRoster, currentRound) as the active roster in the roster state.
```

Comment:
Minor: basically, we want to perform 7 and 8 in order to build the 6. I'd suggest to either make the 7 and 8 sub-points of the 6, or put 6 after the 7 and 8 and state that this history state is the result of the previous two actions. That would look more logical compared to the current way the points are presented.

Reply:
reordered